apps/mobile/src/store/key-store.ts

```diff
@@ -90,11 +92,45 @@ export function useKeyStore() {
       );
     },
 
-    async deriveNextAccountKeychainsFrom(fingerprint: string) {
-      const { mnemonic, passphrase } = await mnemonicStore(fingerprint).getMnemonic();
-
-      if (!mnemonic) throw new Error('No mnemonic found for fingerprint ' + fingerprint);
+    async deriveNextAccountKeychainsFrom(
```

Comment:
How about having two functions. 
- `deriveNextAccountKeychainsFromFingerprint` 
- `deriveNextAccountKeychains` - accepts `mnemonic` + `passphrase` and calls `deriveNextAccountKeychainsFromFingerprint` ?

I think that will make it cleaner and easier to follow

Reply:
+1